core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -857,9 +862,11 @@ private[spark] class BlockManager(
 
     val startTimeMs = System.currentTimeMillis
     var blockWasSuccessfullyStored: Boolean = false
+    var exceptionWasThrown: Boolean = true
     val result: Option[T] = try {
       val res = putBody(putBlockInfo)
       blockWasSuccessfullyStored = res.isEmpty
+      exceptionWasThrown = false
       res
     } finally {
       if (blockWasSuccessfullyStored) {
```

Comment:
Ok this is fine then. Could you leave a comment mentioning the InterruptedException problem ? Otherwise, this LGTM